src/ucp/tag/rndv.c

```diff
@@ -1156,10 +1231,14 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
                 goto out;
             }
         }
-        /* put protocol is allowed - register receive buffer memory for rma */
-        ucs_assert(rndv_rts_hdr->size <= rreq->recv.length);
-        ucp_request_recv_buffer_reg(rreq, ep_config->key.rma_bw_md_map,
-                                    rndv_rts_hdr->size);
+
+        if ((rndv_mode == UCP_RNDV_MODE_PUT_ZCOPY) ||
+            UCP_MEM_IS_CUDA(rreq->recv.mem_type)) {
```

Comment:
why need to check cuda here?

Reply:
in default UCP_RNDV_MOD_AUTO, we try to use PUT protocol for CUDA IPC case.